enzyme/Enzyme/CMakeLists.txt

```diff
@@ -283,15 +283,20 @@ set_target_properties(LLDEnzyme-${LLVM_VERSION_MAJOR} PROPERTIES
         LINK_FLAGS "-Wl,-flat_namespace -Wl,-undefined -Wl,suppress")
 endif()
 
+
 add_library(LLDEnzyme INTERFACE)
 target_compile_options(LLDEnzyme INTERFACE -flto)
 target_link_options(LLDEnzyme INTERFACE "SHELL: -fuse-ld=lld")
-if(${LLVM_VERSION_MAJOR} LESS 15) 
+
+set(LTO_LEGACY_MANAGER_VERSIONS 13 14)
+if (${LLVM_VERSION_MAJOR} IN_LIST LTO_LEGACY_MANAGER_VERSIONS)
   target_link_options(LLDEnzyme INTERFACE "SHELL: -Wl,--lto-legacy-pass-manager")
 endif()
```

Comment:
Per your comment on the PR itself -- I think the LTO plugin also needs to use a different (new passmanger specific) flag on 15+. I don't know what it is offhand, but presumably we can find this in the LLVM docs.